### DISCUSSION TITLE
Javascript canvas support

### DISCUSSION THREAD

**Discussion by tonmoy998** [LazyVim/LazyVim#1353] [Q&A]
Javascript canvas support and custom javascript library support??

**Comment by abeldekat** [comment]
Hello @tonmoy998,

Why are you using "??" ?

**Reply by tonmoy998** [reply]
Hi, I am using javascript 2d canvas and learning three js.How to enable autocomplete function for them?

**Reply by dpetka2001** [reply]
Autocompletion depends largely on the LSP server that you use (usually `tsserver`). Lazyvim has an extra for Typescript, read more in the [docs](https://www.lazyvim.org/extras/lang/typescript) about how to enable it.

**Reply by tonmoy998** [reply]
Thanks for your suggestion
